app/__init__.py

```diff
@@ -101,6 +109,16 @@ def refresh_session():
         session.permanent = True
         session.modified = True
 
+    @application.context_processor
+    def inject_saved_search_temp_message_statuses():
+        return dict(
+            NOT_LOCKED_PRE_LIVE=NOT_LOCKED_PRE_LIVE,
+            NOT_LOCKED_POST_LIVE=NOT_LOCKED_POST_LIVE,
+            LOCKED_PRE_LIVE=LOCKED_PRE_LIVE,
+            LOCKED_POST_LIVE_DURING_INTERIM=LOCKED_POST_LIVE_DURING_INTERIM,
+            LOCKED_POST_LIVE_POST_INTERIM=LOCKED_POST_LIVE_POST_INTERIM,
+        )
```

Comment:
If you change the states in search_save_helpers into an enum, you can simplify this code to avoid the name duplication:


```python
class SavedSearchStateEnum(enum.Enum):
    NOT_LOCKED_PRE_LIVE = 1
    NOT_LOCKED_POST_LIVE = 2
    LOCKED_PRE_LIVE = 3
    LOCKED_POST_LIVE_DURING_INTERIM = 4
    LOCKED_POST_LIVE_POST_INTERIM = 5
```

```python
@application.context_processor
    def inject_saved_search_temp_message_statuses():
        return {state.name: state.value for state in SavedSearchStateEnum}
```

Reply:
Oh nice. I'll change.